shared/modules/ibc_store_module.go

```diff
@@ -0,0 +1,50 @@
+package modules
+
+import (
+	ics23 "github.com/cosmos/ics23/go"
+	"github.com/pokt-network/pocket/persistence/kvstore"
+	"github.com/pokt-network/pocket/runtime/configs"
+	coreTypes "github.com/pokt-network/pocket/shared/core/types"
+)
+
+//go:generate mockgen -destination=./mocks/ibc_store_module_mock.go github.com/pokt-network/pocket/shared/modules BulkStoreCacher,ProvableStore
+
+const BulkStoreCacherModuleName = "bulk_store_cache"
+
+type BulkStoreCacherOption func(BulkStoreCacher)
+
+type bulkStoreCacheFactory = FactoryWithConfigAndOptions[BulkStoreCacher, *configs.BulkStoreCacherConfig, BulkStoreCacherOption]
+
+// BulkStoreCacher is a submodule that interacts with the different stores it manages and handles the
+// flushing, pruning and restoration of their caches in bulk
+type BulkStoreCacher interface {
+	Submodule
+	bulkStoreCacheFactory
+
+	AddStore(name string) error
+	GetStore(name string) (ProvableStore, error)
+	RemoveStore(name string) error
+	GetAllStores() map[string]ProvableStore
+	FlushAllEntries() error
```

Comment:
OPTIONAL: `FlushEntriesFromCachestoStores`

I realize this is a bit pedantic but just TOL (thinking out loud).

@bryanchriswhite wdyt?

Reply:
I like the direction, what about just `FlushCachesToStore`?

Assuming no understanding of what I'm looking at (which is pretty accurate at this point :sweat_smile:), I like `FlushEntriesFromCachesToStores` better than `FlushAllEntries` as it's not clear to me from the interface definition what an "entry" is and/or how it relates to anything else described in the interface. I think `FlushCachesToStore` better captures the minimal relevant context.